### PR TITLE
Add retry logic

### PR DESCRIPTION
**Pull request by josefkarasek**

### CODE REVIEW

**Review by lmzuccarelli — COMMENTED** [review]
small fixes if you don't mind 
1. Define a constant instead of duplicating this literal 3 times i.e "Instance"	r.Log.V(3).Info("Reconcile", "Instance", req.NamespacedName.String())
2. If possible do it for other repeats as well 

For code smell explanation 
```
Duplicated string literals make the process of refactoring error-prone, since you must be sure to update all occurrences.
On the other hand, constants can be referenced from many places, but only need to be updated in a single place.
```

File: pkg/operator/controller/nodeobservabilityrun/error.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+func (e NodeObservabilityRunError) Error() string {
+	return fmt.Sprintf("Code: %d, Error: %s", e.HttpCode, e.Msg)
+}
+
+func IsNodeObservabilityRunErrorRetriable(err error) bool {
+	e, ok := err.(NodeObservabilityRunError)
+	if !ok {
+		return false
+	}
+	if e.HttpCode == 500 {
```
Comment (josefkarasek) [inline pkg/operator/controller/nodeobservabilityrun/error.go:19]: use constant from net/http

File: pkg/operator/controller/nodeobservabilityrun/controller.go [hunk 2]
```diff
@@ -165,23 +168,28 @@ func inProgress(instance *nodeobservabilityv1alpha1.NodeObservabilityRun) bool {
...
+		defer resp.Body.Close()
 
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return "", 0, err
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode != 200 {
```
Comment (josefkarasek) [inline pkg/operator/controller/nodeobservabilityrun/controller.go:188]: use constant

File: pkg/operator/controller/nodeobservabilityrun/error.go (hunk 1 above)
Comment (sherine-k) [reply pkg/operator/controller/nodeobservabilityrun/error.go:19]: when the agent returns 500, in most cases, with the current implementation, it is "stuck" until an admin cleans up the error file.
I don't know if we want to keep retrying in this case.
Do we want to look at httpCode 409? It's returned when the agent is busy by another profiling request, and will be able to run this request a bit later.

**Pull request merged**